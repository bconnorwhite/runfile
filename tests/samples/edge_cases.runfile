# ----------
# Edge Cases
# ----------

# Very long command name
verylongcommandnamethatexceedstypicallimits
  shortarg?  # Short argument name
  --verylongflagnamethatexceedstypicallimits: # Very long flag name
  echo "Running very long command"

# Command with special characters in names
test-special_chars
  arg_with_underscores?     # Argument with underscores
  --flag-with-dashes        # Flag with dashes
  --flag_with_underscores:  # Flag with underscores
  echo "Special chars: $arg_with_underscores"

# Command with minimal description
minimal
  a?    # a
  --b:  # b
  echo "minimal"

# Command with very long descriptions
verbose
  argument? # This is a very long description that should test how the alignment works when we have extremely long descriptions that might wrap or cause formatting issues
  --flag:   # Another very long description that should test the alignment and formatting capabilities of the help system when dealing with extremely verbose documentation
  echo "verbose command"

# Command with no description
nodocs
  arg
  --flag:
  echo "no docs"

# Command with mixed argument types
mixed
  required_arg       # Required argument
  optional_arg?      # Optional argument
  -r, --release      # Short and long flag
  --debug            # Long flag only
  -v, --verbose      # Another short and long flag
  --output=<file>:   # Flag with value
  echo "Mixed args: $required_arg $optional_arg"
  echo "Flags: release=$release debug=$debug verbose=$verbose output=$output"
