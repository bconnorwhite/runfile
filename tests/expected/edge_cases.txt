Edge Cases
  verylongcommandnamethatexceedstypicallimits   # Very long command name
    shortarg?                                     # Short argument name
    --verylongflagnamethatexceedstypicallimits    # Very long flag name
  test-special_chars                            # Command with special characters in names
    arg_with_underscores?                         # Argument with underscores
    --flag-with-dashes                            # Flag with dashes
    --flag_with_underscores                       # Flag with underscores
  minimal                                       # Command with minimal description
    a?                                            # a
    --b                                           # b
  verbose                                       # Command with very long descriptions
    argument?                                     # This is a very long description that should test how the alignment works when we have extremely long descriptions that might wrap or cause formatting issues
    --flag                                        # Another very long description that should test the alignment and formatting capabilities of the help system when dealing with extremely verbose documentation
  nodocs                                        # Command with no description
    arg
    --flag
  mixed                                         # Command with mixed argument types
    required_arg                                  # Required argument
    optional_arg?                                 # Optional argument
    -r, --release                                 # Short and long flag
    --debug                                       # Long flag only
    -v, --verbose                                 # Another short and long flag
    --output                                      # Flag with value
